diff --git a/includes/media.fields.inc b/includes/media.fields.inc
index e0c4bbc..f9fb7e8 100644
--- a/includes/media.fields.inc
+++ b/includes/media.fields.inc
@@ -392,7 +392,7 @@ function theme_media_widget($variables) {
   $output = '';
 
   // The "form-media" class is required for proper Ajax functionality.
-  $output .= '<div class="media-widget form-media clearfix">';
+  $output .= '<div id="' . $element['#original_id'] . '" class="media-widget form-media clearfix">';
   $output .= drupal_render_children($element);
   $output .= '</div>';
 
diff --git a/media.module b/media.module
index 1f97bcb..e7ec4f5 100644
--- a/media.module
+++ b/media.module
@@ -715,7 +715,7 @@ function media_element_process($element, &$form_state, $form) {
 
   // Append the '-upload' to the #id so the field label's 'for' attribute
   // corresponds with the textfield element.
-  $original_id = $element['#id'];
+  $original_id = $element['#original_id'] = $element['#id'];
   $element['#id'] .= '-upload';
   $fid = isset($element['#value']['fid']) ? $element['#value']['fid'] : 0;
 
